debug: cpu_load: Extend support to RISCV architecture and nrf54h20

nordic-krch opened the pull request:
Add `sys_trace_idle_exit` to RISCV cpu idle functions.
Add `sys_trace_idle` and `sys_trace_idle_exit` to nrf54h20 specific idle states.

Fixes #87090.

nordic-krch committed:
soc: nordic: nrf54h: power: Add idle tracing calls

Add calls to sys_trace_idle and sys_trace_idle_exit in nrf54h specific
idle states to allow measuring CPU load on nrf54h20 when power
management is enabled.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

masz-nordic commented on subsys/debug/Kconfig:
```diff
@@ -435,7 +435,7 @@ config CS_TRACE_DEFMT
 config CPU_LOAD
 	select TRACING
 	# Currently hooks are added only to Cortex M architecture.
```
```suggestion
```

nashif commented:
how did the test tests/subsys/debug/cpu_load manage to run on riscv? it is filtered for cortex_m!!

fkokosinski reviewed (changes requested):
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/subsys/debug/cpu_load/testcase.yaml needs adjusting to allow RISC-V arch

henrikbrixandersen reviewed (commented):
I would split commit titled "arch: riscv: Add support for CPU load measuring" into two; one commit for adding CPU load monitoring to the RISC-V arch, and one for changing the test suite.

nordic-krch committed:
arch: riscv: Add support for CPU load measuring

Add sys_trace_idle_exit to RISCV cpu_idle functions and allow
enabling CPU_LOAD module for RISCV and disable it for SMP.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: debug: cpu_load: Extend filtering with riscv

Allow test execution for riscv since it now supports cpu_load module.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch commented:
> I would split commit titled "arch: riscv: Add support for CPU load measuring" into two; one commit for adding CPU load monitoring to the RISC-V arch, and one for changing the test suite.

@henrikbrixandersen  Done.
